tests/queries/0_stateless/03307_loop_table_function_as_engine.sh

```diff
@@ -0,0 +1,14 @@
+#!/usr/bin/env bash
```

Comment:
In #73804, the test is called `[...]_forbid_cluster_table_engine.sh`. We should include `forbid` in the test name as well.

(Besides that, I often point to [minimal reproducible examples](https://en.wikipedia.org/wiki/Minimal_reproducible_example) in user questions but the same principle applies to tests as well. What is the essence of the test? --> Usage of the `Loop` table engine. This raises two questions:
- Why does the test table use three columns? A single test column would be just fine, any further columns and complexity distract from the purpose of the test.
- About complexity: Why is this not a .sql test? Something like `CREATE TABLE tab (col String) ENGINE=Loop; -- { serverError BAD_ARGUMENTS }` would work just fine. (the same argument could be made in #73804)
)

Reply:
Also like to convert `03293_forbid_cluster_table_engine.sh` to SQL? (can also be done in a separate PR)